cmd/ctr/commands/tasks/resource.go

```diff
@@ -0,0 +1,92 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package tasks
+
+import (
+	"errors"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cmd/ctr/commands"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/urfave/cli"
+)
+
+func init() {
+	// resourceCommand is only added on Linux, does not compile on darwin or windows
+	Command.Subcommands = append(Command.Subcommands, resourceCommand)
+}
+
+var resourceCommand = cli.Command{
+	Name:           "resource",
+	Usage:          "change resource limits of existing container",
+	ArgsUsage:      "[flags] CONTAINER",
+	SkipArgReorder: true,
+	Flags: []cli.Flag{
+		cli.Uint64Flag{
+			Name:  "memory-limit",
+			Usage: "memory limit (in bytes) for the container",
+		},
+		cli.Int64Flag{
+			Name:  "cpu-quota",
+			Usage: "Limit CPU CFS quota",
+			Value: -1,
+		},
+		cli.Uint64Flag{
+			Name:  "cpu-period",
+			Usage: "Limit CPU CFS period",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		var (
+			id     = context.Args().First()
+			limit  = context.Int64("memory-limit")
+			quota  = context.Int64("cpu-quota")
+			period = context.Uint64("cpu-period")
+		)
+		if id == "" {
+			return errors.New("container id must be provided")
+		}
+		client, ctx, cancel, err := commands.NewClient(context)
+		if err != nil {
+			return err
+		}
+		defer cancel()
+		container, err := client.LoadContainer(ctx, id)
+		if err != nil {
+			return err
+		}
+
+		task, err := container.Task(ctx, nil)
+		if err != nil {
+			return err
+		}
+
+		if err := task.Update(ctx, containerd.WithResources(&specs.LinuxResources{
+			Memory: &specs.LinuxMemory{
+				Limit: &limit,
```

Comment:
@AkihiroSuda 
I didn't understand the comment well. Please let me know in more detail.
You mean that how does it work when limit is unspecified?
Then it make error message like "ctr: flag needs an argument: -memory-limit".

Reply:
When the flag wasn't specified, this code seems to set the limit value to zero, but it should set nil

Reply:
@AkihiroSuda 
hmm... It works well... 
on containerd/cgroup check the value verification,
I'm curious about your opinion whether it's better to take the defaults here or  entrust it to the cgroup as it is now.

Reply:
@crisesw I think it is best to not change unless it is necessary and intentional.

Reply:
I agree with @AkihiroSuda and @cpuguy83. If the user of this command doesn't specify values explicitly, the command shouldn't change them.

Reply:
When I think about it again, I agree with everyone's opinion.
I wanted to implement a way to increase/decrease memory without killing the running task. But I think it would be right to do it exactly when starting out explicitly.
We will close this pull request.